### ISSUE TITLE
project still active?

### ISSUE THREAD

**Issue by keikoro** [sarahgp/p5bots#33]
Based on unanswered issues and unmerged PRs, I'm wondering if this project is still being actively maintained? (Was thinking of sending a PR for the README but not sure how much sense it makes at this point.)

**Comment by keikoro** [comment]
Addendum: as the lib is listed at https://p5js.org/libraries/, which gives it somewhat official backing, perhaps it would make sense to move it under the @processing foundation umbrella to help with maintainance?

**Comment by monteslu** [comment]
now that we have webUSB and webSerial we don't really need the node server backend.
I've got a p5 plugin that works with any microcontroller that can run firmata:  https://github.com/monteslu/p5.j5

**Comment by keikoro** [comment]
I feel like I'm lacking context, who is "we" here? Are you involved in this repo?

**Comment by monteslu** [comment]
> I feel like I'm lacking context, who is "we" here? Are you involved in this repo?

WebSerial is an open spec with an implementation in Chrome, Edge, Chromium, and Brave.

I helped the spec authors and implementers in testing webSerial, webUSB, and webBluetooth.  

P5.J5 is just something I wrote which is similar to this project, but is completely browser based and doesn't require running node.js.  It also has all the features of this (firmata) plus all of [johnny-five](https://github.com/rwaldron/johnny-five) and some additional helper libraries.  

More info here:  https://twitter.com/monteslu/status/1283439166408155136

**Comment by keikoro** [comment]
@monteslu Picking up on this again: I originally found my way here via the official p5.js website, which has this project listed on its [libraries](https://p5js.org/libraries/) page as _the only_ one for using p5 with Arduino. Other tutorials etc. I found via googling also only ever mentioned p5bots.

I don't know (about) any of the things you mentioned here, it basically all sounds like jargon I'm unfamiliar with to me. I don't know what these tools are, if they are p5 only, or general JavaScript/browser libraries or plugins or... (i.e. if these are things I should have heard of, or if my lack of p5.js knowledge is to blame for not having a clue).

So basically everything was pointing to this being the library I was looking for, but having had a look at the repo, the project seemed abandoned to me. Now I don't know if it is but it's still the defacto standard for p5 + Arduino work, or if everyone has moved on to something else but no-one bothered to update their websites, or make blog posts about it or whatever.

You linking to something you built yourself (which seems rather new and doesn't seem to have many users yet, based on the faves) isn't particularly helpful to someone like me who's unfamiliar with the ecosystem.

**Comment by monteslu** [comment]
@keikoro no worries.  Just a suggestion since this one seems abandoned.  I'm sure there's a way to get mine listed on the p5 page, but not sure what that is yet.  I'd love to collaborate with @sarahgp on this repo since the end goals are the same, I think.

I've been involved in javascript robotics (johnny-five, nodebots, nodecopter, nodeboats) and running educational robotics events for the last several years.  Last year some friends and I ran a new event called NodeArt at JSConf: https://2019.jsconf.us/about/team which featured a good intersection of creative coding and electronics.  That type of thing seems to have some overlap with p5, so I thought my plugin would be helpful going forward.

Browser hardware APIs are new, but they're pretty stable these days.  And communicating directly to a classic arduino from a browser is a relatively recent development:  https://twitter.com/monteslu/status/1163966447623688193


My plugin simply makes it easy to use p5 with the open source http://johnny-five.io/ project.

**Comment by sarahghp** [comment]
Thanks for pinging me, @monteslu. This project is no longer active — I've become too busy and let it fall by the wayside. But this is a good reminder that I should hand it off more deliberately. 

In the meantime, you can probably open an issue on the [p5.js website repo](https://github.com/processing/p5.js-website) to add your project, which makes a lot of sense to me.
